src/EnergyPlus/WeatherManager.cc

```diff
@@ -9117,7 +9117,8 @@ namespace WeatherManager {
                 } else {
                     // Environment crosses year boundaries
                     env.RollDayTypeOnRepeat = runPer.RollDayTypeOnRepeat;
-                    env.StartJDay = General::OrdinalDay(runPer.startMonth, runPer.startDay, runPer.isLeapYear ? 1 : 0);
+                    env.StartJDay = General::OrdinalDay(
+                        runPer.startMonth, runPer.startDay, isLeapYear(env.StartYear) && state.dataWeatherManager->WFAllowsLeapYears ? 1 : 0);
```

Comment:
This change means that `env.IsLeapYear`/`runPer.isLeapYear` aren't definitive on whether the first year of the environment is a leap year or not, so this is not going to work by itself. The reported issue needs more information before it's at all clear what the fix is.